lab-steven/lib/text-read.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const fs = require('fs');
+
+// module.exports = exports = {};
+
+
+
+// exports.readDone = [];
+
+module.exports.textReader = function(filename1, filename2, filename3, callback) {
+  let fileReadDone = [];
+  fs.readFile(`${__dirname}/../data/${filename1}.txt`, function(err, data){
+    if(err) throw err;
+    // console.log(data.toString('hex', 0, 8), filename);
+    fileReadDone.push(data.toString('hex', 0, 8));
+    // console.log(fileReadDone);
+
+    fs.readFile(`${__dirname}/../data/${filename2}.txt`, function(err, data){
+      if(err) throw err;
+      // console.log(data.toString('hex', 0, 8), filename);
+      fileReadDone.push(data.toString('hex', 0, 8));
+      // console.log(fileReadDone);
+
+      fs.readFile(`${__dirname}/../data/${filename3}.txt`, function(err, data){
+        if(err) throw err;
+        // console.log(data.toString('hex', 0, 8), filename);
+        fileReadDone.push(data.toString('hex', 0, 8));
+        // console.log(fileReadDone);
+        // return callback(fileReadDone);
+        exports.readDone.push(file)
```

Comment:
where is "file" coming from? im assuming you want to pass the contents of your array in here? that's why it's not accessible in your test file. find a way to pass the contents to the exports.readDone property so that it's accessible in your test, and you should be solid. the pushes are nested, so it should read asynchronously.